src/test/java/CollectorTest.java

```diff
@@ -0,0 +1,113 @@
+    @AfterAll
+    static void afterAll() {
+        System.setSecurityManager(null);
+    }
+
+    /** Custom exception to be thrown whe never {@link System#exit(int)} is invoked. */
+    private static class ExitException extends SecurityException {
+        public final int status;
+
+        public ExitException(int status) {
+            this.status = status;
+        }
+    }
+
+    /** Override {@link SecurityManager} to prevent exiting JVM */
+    private static class DoNotExitJVM extends SecurityManager {
+        @Override
+        public void checkPermission(Permission perm) {}
+
+        @Override
+        public void checkPermission(Permission perm, Object context) {}
+
+        /**
+         * Throws an {@link ExitException} instead of exiting the JVM.
+         *
+         * @param status Exit code of the invocation
+         */
+        @Override
+        public void checkExit(int status) {
+            super.checkExit(status);
+            throw new ExitException(status);
+        }
+    }
+
+    @Test
+    void should_exitWithZero_withNonEmptyOutput(@TempDir Path tempDir) throws IOException {
+        // arrange
+        Path outputJson = tempDir.resolve("output.json");
+        String classpath =
+                TestHelper.getMavenClasspathFromBuildDirectory(
+                        TestHelper.PATH_TO_SAMPLE_MAVEN_PROJECT.resolve("with-debug"));
+        String[] args = {
+            "-i",
+            TestHelper.PATH_TO_SAMPLE_MAVEN_PROJECT.resolve("input.txt").toString(),
+            "-p",
+            classpath,
+            "-t",
+            TestHelper.PATH_TO_SAMPLE_MAVEN_PROJECT.resolve("src/test/").toString(),
+            "-o",
+            outputJson.toString(),
+        };
+
+        // assert
+        ExitException exit = assertThrows(ExitException.class, () -> Collector.main(args));
+        assertEquals(0, exit.status);
+
+        Reader reader = new FileReader(outputJson.toFile());
+        int fileSize = reader.read();
+        assertNotEquals(-1, fileSize);
+    }
+
+    @Test
+    void should_exitWithNonZeroCode(@TempDir Path tempDir) {
+        // arrange
+        Path outputJson = tempDir.resolve("output.json");
+        String classpath =
+                TestHelper.getMavenClasspathFromBuildDirectory(
+                        TestHelper.PATH_TO_SAMPLE_MAVEN_PROJECT.resolve("without-debug"));
+        String[] args = {
+            "-i",
+            TestHelper.PATH_TO_SAMPLE_MAVEN_PROJECT.resolve("input.txt").toString(),
+            "-p",
+            classpath,
+            "-t",
+            TestHelper.PATH_TO_SAMPLE_MAVEN_PROJECT.resolve("src/test/").toString(),
+            "-o",
+            outputJson.toString(),
+        };
+
+        // assert
+        ExitException exit = assertThrows(ExitException.class, () -> Collector.main(args));
+        assertNotEquals(0, exit.status);
+
+        assertThat(outputJson.toFile(), not(anExistingFile()));
```

Comment:
@slarse, I needed some help with this test case. You don't need to read what it is doing.

The main method exits with `1` because an instance of `AbsentInformationException` is thrown. Any idea how I can check for the exception instead?

Reply:
> The main method exits with 1 because an instance of AbsentInformationException is thrown. Any idea how I can check for the exception instead?

I can't see that happening anywhere, for me (and GitHub Actions) the tests don't even start to run. This is the error I see ("wrong name"): https://github.com/algomaster99/collector-sahab/runs/5632590271?check_suite_focus=true#step:5:5100

When running with IntelliJ IDEAs test runner, the tests work fine, so it seems to me like this is a misconfiguration of some form, or potentially that you're doing something that Surefire doesn't support.

Reply:
Yes, I saw that. Somehow the tests are passing if I run it through IntelliJ junit runner. However, `mvn tests` fails to execute any tests.

Reply:
@slarse I have explicitly configured in `pom.xml` which tests I want to run so the CI is working now.  I have plans to come up with a more generic solution for this, but as of now, I want to add some functionalities before I spend more on chores.

Could you please have a look at my question again?
> The main method exits with 1 because an instance of AbsentInformationException is thrown. Any idea how I can check for the exception instead?

It happens [here](https://github.com/algomaster99/collector-sahab/runs/5660372357?check_suite_focus=true#step:5:7591) in the latest workflow run.